src/schemas/json/jj.json

```diff
@@ -0,0 +1,136 @@
+        },
+        "ui": {
+            "type": "object",
+            "description": "UI settings",
+            "properties": {
+                "allow-init-native": {
+                    "type": "boolean",
+                    "description": "Whether to allow initializing a repo with the native backend"
+                },
+                "relative-timestamps": {
+                    "type": "boolean",
+                    "description": "Whether to change timestamps to be rendered as a relative description instead of a full timestamp",
+                    "default": false
+                },
+                "default-revset": {
+                    "type": "string",
+                    "description": "Default set of revisions to show when no explicit revset is given for jj log and similar commands",
+                    "default": "@ | (remote_branches() | tags()).. | ((remote_branches() | tags())..)-"
+                },
+                "color": {
+                    "type": "string",
+                    "description": "Whether to colorize command output",
+                    "enum": ["always", "never", "auto"],
+                    "default": "auto"
+                },
+                "pager": {
+                    "type": "string",
+                    "description": "Pager to use for displaying command output",
+                    "default": "less -FRX"
+                },
+                "editor": {
+                    "type": "string",
+                    "description": "Editor to use for commands that involve editing text"
+                },
+                "diff-editor": {
+                    "type": "string",
+                    "description": "Editor tool to use for editing diffs",
+                    "default": "meld"
+                },
+                "merge-editor": {
+                    "type": "string",
+                    "description": "Tool to use for resolving three-way merges. Behavior for a given tool name can be configured in merge-tools.TOOL tables"
+                }
+            }
+        },
+        "colors": {
+            "type": "object",
+            "description": "Mapping from jj formatter labels to ANSI color names"
+        },
+        "merge-tools": {
+            "type": "object",
+            "description": "Tables of custom options to pass to the given merge tool (selected in ui.merge-editor)",
+            "additionalProperties": {
+                "type": "object",
+                "properties": {
+                    "program": {
+                        "type": "string"
+                    },
+                    "merge-args": {
+                        "type": "array",
+                        "items": {
+                            "type": "string"
+                        }
+                    },
+                    "merge-tool-edits-conflict-markers": {
+                        "type": "boolean",
+                        "default": false
+                    }
```

Comment:
FYI, there's a pretty detailed explanation [here](https://github.com/martinvonz/jj/blob/main/docs/config.md#editing-conflict-markers-with-a-tool-or-a-text-editor). Do we want to link to that from here? If we have the link at the top, it might not be necessary, but maybe that depends on how tools generally present the descriptions.

I don't know how we want to deal with the redundancy between this file and the docs in the long term. We'll probably have to accept that there will be some redundancy.

Reply:
Sure, I'll link.

Mostly I was keeping descriptions terse because I didn't like the long lines and didn't want to dig into how to nicely format multiline descriptions just yet. I'll go ahead and figure that out.

Reply:
And re redundancy, yes some is unavoidable IMO. For the bulk of redundancy we could set up some scripts to generate markdown reference docs from parts of the schema, though.

Reply:
Added a quick description & link to the markdown.